packages/apps/fortune/exchange-oracle/server/src/modules/job/job.service.ts

```
@@ -0,0 +1,98 @@
+  ) {}
+
+  public async getDetails(
+    chainId: number,
+    escrowAddress: string,
+  ): Promise<JobDetailsDto> {
+    const signer = this.web3Service.getSigner(chainId);
+    const escrowClient = new EscrowClient(await InitClient.getParams(signer));
+    const manifestUrl = await escrowClient.getManifestUrl(escrowAddress);
+    const manifest = await StorageClient.downloadFileFromUrl(manifestUrl);
+
+    return {
+      escrowAddress,
+      chainId,
+      manifest: {
+        title: manifest.title,
+        description: manifest.description,
+        fortunesRequested: manifest.fortunesRequired,
+        fundAmount: manifest.fundAmount,
+      },
+    };
+  }
+
+  public async getPendingJobs(
+    chainId: number,
+    workerAddress: string,
+  ): Promise<string[]> {
+    const signer = this.web3Service.getSigner(chainId);
+    const escrowClient = new EscrowClient(await InitClient.getParams(signer));
+    const escrows = await escrowClient.getEscrowsFiltered({
+      status: EscrowStatus.Pending,
+    });
+
+    return escrows.filter(
+      (escrow) => !this.storage[escrow]?.includes(workerAddress),
+    );
+  }
+
+  public async solveJob(
+    chainId: number,
+    escrowAddress: string,
+    workerAddress: string,
+    solution: string,
+  ): Promise<boolean> {
+    const signer = this.web3Service.getSigner(chainId);
+    const escrowClient = new EscrowClient(await InitClient.getParams(signer));
+    const recordingOracleAddress = await escrowClient.getRecordingOracleAddress(
+      escrowAddress,
+    );
+    const kvstore = new KVStoreClient(await InitClient.getParams(signer));
+    const recordingOracleURL = await kvstore.get(
+      recordingOracleAddress,
+      KVStoreKeys.webhook_url,
+    );
+
+    if (!this.storage[escrowAddress]) {
+      this.storage[escrowAddress] = [];
+    }
+    this.storage[escrowAddress].push(workerAddress);
+
+    await this.httpService.post(
+      (recordingOracleURL || 'http://localhost:3005') + '/job/solve',
```

if recordingOracleURL is null it should throw an error